### PR TITLE
Record batch

### PR DESCRIPTION
**Pull request by JohanMabille**

### CODE REVIEW

File: include/sparrow/record_batch.hpp [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+{
+    /**
+     * Table-like data structure.
+     *
+     * A record batch is a collection of equal-length arrays mapped to
+     * names.
+     *
+     * @tparm T the type of the names
+     */
+    template <class T = std::string>
```
Comment (DerThorsten) [inline include/sparrow/record_batch.hpp:37]: do we really need string as a template here?

File: src/array.cpp [hunk 2]
```diff
@@ -81,5 +81,23 @@ namespace sparrow
...
+        {
+            return rhs.visit([&typed_lhs](const auto& typed_rhs) -> bool
+            {
+                if constexpr (!std::same_as<decltype(typed_lhs), decltype(typed_rhs)>)
+                {
+                    return false;
+                }
+                else
+                {
+                    return typed_lhs == typed_rhs;
```
Comment (DerThorsten) [inline src/array.cpp:97]: I am a bit puzzled by this. Is == implemented by each layout?

File: include/sparrow/record_batch.hpp (hunk 1 above)
Comment (JohanMabille) [reply include/sparrow/record_batch.hpp:37]: Probably not, I'll remove it.

File: src/array.cpp (hunk 2 above)
Comment (JohanMabille) [reply src/array.cpp:97]: Layouts inheriting from `array_crtp_base` get it for free. The other layouts must implement it, I'll check if it's the case for each of them.

File: test/test_record_batch.cpp
Comment (Alex-PLACET) [inline test/test_record_batch.cpp:1]: Some test are missings for:
size_type nb_columns() const;
size_type nb_rows() const;
name_range names() const;
column_range columns() const;

Comment (JohanMabille) [reply test/test_record_batch.cpp:1]: The missing tests have been added.

File: src/array.cpp (hunk 2 above)
Comment (JohanMabille) [reply src/array.cpp:97]: Checked, all the layouts inherit or implement `operator==`.

**Review by Alex-PLACET — APPROVED** [review]

**Review by DerThorsten — APPROVED** [review]

**Pull request merged**